### ISSUE TITLE
force absolute urls

### ISSUE THREAD

**Issue by Strat-** [retrospring/bugs#79]
[woo, markdown](https://retrospring.net/str_t/a/104964)

*for those who don't get it, click on the question's url*

**Comment by yukimono** [comment]
it's per design of both markdown and html.

``<a href="google.com"></a>``
directs to 
http://example.com/google.com
``<a href="//google.com"></a>``
directs to 
http://google.com

the caveat is that you *cant* so a simple check on the URL to see if it's a sane address, because if that would be the case relative URLs have a tendency to just break, unless you add more "is it sane" checks, which ultimately adds more processing to a human error.

tl;dr
it's by design, not a bug.

**Comment by yukimono** [comment]
it's per design of both markdown and html.

``<a href="google.com"></a>`` or [[google.com]](google.com)
directs to  
http://example.com/google.com
``<a href="//google.com"></a>`` or [[//google.com]](//google.com)
directs to 
http://google.com

the caveat is that you *cant* so a simple check on the URL to see if it's a sane address, because if that would be the case relative URLs have a tendency to just break, unless you add more "is it sane" checks, which ultimately adds more processing to a human error.

tl;dr
it's by design, not a bug.

**Comment by Strat-** [comment]
@yukimono I know how Markdown does it. I'm saying that they need to override the behavior.

**Comment by yukimono** [comment]
So what you're saying is make every single URL absolute regardless of the context

**Comment by yukimono** [comment]
The issue that was specific to that question's formatting was because he had manually entered a markdown link, at the top level of markdown there's a simple URL preprocessor that catches most urls even without protocol prefixes.

So again

Human error

**Comment by thebluecorsair** [comment]
If a /bug/ is caused by human error, there are always ways to make it less common to occur, however as this is how Markdown was designed (the same rules apply to BBCode, etc), I don't think forcing an a URL change should be necessary.

This is just a scenario where the user should notice the URL was inputted incorrectly and either delete+edit it, or simply learn for the next time. There isn't much to resolve in this case.

**Comment by yukimono** [comment]
The bug only happens when two things have to happen.

1. the user needs to understand how to add markdown anchor tags, and 
2. the user needs to forget the protocol tag

The likelihood of anyone who doesn't know about 2 while knowing of 1 is kinda eh. Markdown automatically adds protocols if you avoid the anchor tag.

**Comment by Strat-** [comment]
@yukimono 

>So what you're saying is make every single URL absolute regardless of the context

Yes, precisely that. Regular people don't know (and don't give two shits) about the difference between `http://google.com` and `google.com`, much less us expecting them to use relative URL.s

**Comment by yukimono** [comment]
but my stance is why would they add in a markdown tag they probably don't care or don't know about anyway?

**Comment by Strat-** [comment]
@yukimono Good point.

Like BBCode, people can eventually learn how to embed URLs (or they can click on a "add link button", but that's its own issue), but it's less likely that they'll learn the difference between `google.com` and `http://google.com`.
